### PR TITLE
[DOCS] DSB-321: Add workflow diagram to OSS Quickstart

### PR DESCRIPTION
**Pull request by rachhouse**
This PR adds a simple workflow diagram (defined via Mermaid markdown!) that provides users with an end-to-end visual of the steps accomplished in the Quickstart. I also added TechnicalTags to the page so that GX terms get visually identified for users and they can benefit from the mouseover tooltips.

https://greatexpectations.atlassian.net/browse/DSB-321

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by rachhouse** [comment]
Direct link to the Quickstart changes preview:
https://deploy-preview-8804.docs.greatexpectations.io/docs/tutorials/quickstart/

**Review by HaebichanGX — APPROVED** [review]
LGTM for first push. One thing I thing I would like to confirm is if we'd like to push our end users to use validators or validations officially. When we run validators and save expectation suites, it creates expectation suite called "default" every time you run the code, all expectations will be saved to the same "default" expectation suite.

**Review by joshzzheng — APPROVED** [review]
Overall looks great!

Some non-blocking feedback/thoughts for future iteration.
- Echoing @HaebichanGX's thoughts on the validator workflow.
- Any chance we can add some top padding to the boxes so it's even with bottom?
- With this diagram, I think we'er ready to combine Quickstart into a single snippet?
- Ideally we have some text in the diagram a matches next in ToC, we want to help our users understand the ToC on the left is steps in an overall workflow.

**Comment by rachhouse** [comment]
@HaebichanGX @joshzzheng Thanks for the reviews!

* Yes, let's discuss how we want to change the Quickstart content (Validators/no Validators, multiple snippets vs. one code block), but I agree we don't have to hold up this PR. Updates to the Quickstart can easily be another PR once we decide what we'd like to do.
* I did attempt to match the step containers to the TOC sections, with the exception of DEFINE (we updated from CREATE in my original draft of the diagram based on comments from Rob). I'm fine with either CREATE or DEFINE as the container header.
   *  CONFIGURE -> Configure your GX environment
   * CONNECT -> Connect to source data
   * DEFINE -> Create Expectations
   * VALIDATE -> Validate Data
* @joshzzheng I'll ping you on slack about the padding question, just want to make sure I understand what you're looking for. If it is something that is possible w/ Mermaid, I'll do it as part of this PR.

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md [hunk 1]
```diff
@@ -30,6 +31,55 @@ Windows support for the open source Python version of GX is currently unavailabl
 - pip
 - An internet browser
 
+## Quickstart Workflow
+
+This quickstart walks you through the steps required to implement a simple, end-to-end GX workflow.
```
Comment (kwcanuck) [inline docs/docusaurus/docs/tutorials/quickstart/quickstart.md]:
```suggestion
This quickstart walks you through the steps required to implement a simple, end-to-end GX workflow.
```
This topic already contains an introductory statement. This statement should introduce the diagram and its purpose. For example:

The following diagram illustrates the GX  data validation workflow. 

I would suggest incorporating this diagram into the overview or adding it before the **Prerequisites** section.

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md [hunk 2]
```diff
@@ -30,6 +31,55 @@ Windows support for the open source Python version of GX is currently unavailabl
 - pip
 - An internet browser
 
+## Quickstart Workflow
```
Comment (kwcanuck) [inline docs/docusaurus/docs/tutorials/quickstart/quickstart.md]:
```suggestion
## Data validation workflow
```
Our standard for headings is sentence case.

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md [hunk 3]
```diff
@@ -30,6 +31,55 @@ Windows support for the open source Python version of GX is currently unavailabl
...
+click Configure_B "https://docs.greatexpectations.io/docs/guides/setup/configuring_data_contexts/instantiating_data_contexts/instantiate_data_context/"
+click Connect_A "https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/connect_to_data_lp"
+click Connect_B "https://docs.greatexpectations.io/docs/guides/connecting_to_your_data/connect_to_data_lp"
+click Define_A "https://docs.greatexpectations.io/docs/terms/validator"
+click Define_B "https://docs.greatexpectations.io/docs/guides/expectations/expectations_lp"
+click Validate_A "https://docs.greatexpectations.io/docs/guides/validation/checkpoints/checkpoint_lp"
+click Validate_B "https://docs.greatexpectations.io/docs/terms/data_docs"
+```
+
+*Click on any individual workflow step to explore additional documentation for that step.*
```
Comment (kwcanuck) [inline docs/docusaurus/docs/tutorials/quickstart/quickstart.md]:
```suggestion
Click a workflow step to view related documentation.
```
image
image

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md [hunk 4]
```diff
@@ -44,43 +94,45 @@ Windows support for the open source Python version of GX is currently unavailabl
...
 ## Connect to Data
 
-- Run the following command to connect to existing `.csv` data stored in the `great_expectations` GitHub repository:
+- Run the following command to connect to existing `.csv` data stored in the `great_expectations` GitHub repository and create a <TechnicalTag tag="validator" text="Validator"/> object:
 
     ```python name="tutorials/quickstart/quickstart.py connect_to_data"
     ```
 
-    The example code uses the default Data Context Data Source for Pandas to access the `.csv` data in the file at the specified `path`.
+    This example code uses the default <TechnicalTag tag="data_context" text="Data Context"/> <TechnicalTag tag="datasource" text="Data Source"/> for Pandas to access the `.csv` data from the file at specified URL path.
```
Comment (kwcanuck) [inline docs/docusaurus/docs/tutorials/quickstart/quickstart.md]:
```suggestion
    The code example uses the default <TechnicalTag tag="data_context" text="Data Context"/> <TechnicalTag tag="datasource" text="Data Source"/> for Pandas to access the `.csv` data from the file at the specified URL path.
```

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md (hunk 3 above)
Comment (kwcanuck) [reply docs/docusaurus/docs/tutorials/quickstart/quickstart.md]: I recommend incorporating this text into the introductory statement.

**Review by kwcanuck — CHANGES_REQUESTED** [review]
I recommend simplifying the diagram. Remove the grey background, change the category titles to title case and avoid ALL CAPITALS, change the orange to a light grey, increase the font size (if feasible), change _ValidationResults_ to _Validation Results_.  Have you tested this on a mobile device to determine how it renders? See https://docs.astronomer.io/astro/set-up-ci-cd for an example of a diagram with limited color.

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md (hunk 1 above)
Comment (rachhouse) [reply docs/docusaurus/docs/tutorials/quickstart/quickstart.md]: Sounds good - leading sentence edited and section moved to before **Prerequisites**.

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md (hunk 2 above)
Comment (rachhouse) [reply docs/docusaurus/docs/tutorials/quickstart/quickstart.md]: Updated!

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md (hunk 3 above)
Comment (rachhouse) [reply docs/docusaurus/docs/tutorials/quickstart/quickstart.md]: Done!

File: docs/docusaurus/docs/tutorials/quickstart/quickstart.md (hunk 4 above)
Comment (rachhouse) [reply docs/docusaurus/docs/tutorials/quickstart/quickstart.md]: Fixed!

**Comment by rachhouse** [comment]
Hi @kwcanuck, thanks for your review and comments! I've incorporated your suggestions.

With regard to diagram simplification, I condensed the previously separate "Add a **Data Source**" and "Add a **Data Asset**" into a single step, "Connect to Data" (do you think Data should be **bolded** on that step?).

The Configure -> Connect -> Define -> and Validate blocks were intended to help users relate the workflow steps to the organization of the TOC, which roughly follows that of a GX workflow. Removing them:
* gives us extra space to increase the font size and improve readability of the steps (👍 )
* means we're not including content designed to improve user intuition and understanding of the building blocks of GX workflows and our public docs organization (👎  )

I'd ideally like to include these blocks & headings, but given mermaid's rendering, I think the tradeoff favors the current iteration due to much better sizing & readability. @joshzzheng I know you're also interested in improving user understanding of the public docs organization, do you have an opinion here?

**Comment by kwcanuck** [comment]
@rachhouse the simplified diagram is easier to to review. I understand your concern about removing the Configure -> Connect -> Define -> and Validate blocks. You could add them back if you think they're helpful and you use a color that doesn't violate contrast concerns. Again, I think what you've provided currently improves readability and makes it easy to understand the process and the sequence. 

A few thoughts:

- It's wordy, but _Connect to Data_ could be _Connect to a Data Asset on a Data Source_ with relevant links. If you keep _Connect to Data_, _data_ shouldn't be capitalized. 

- Typically, links to source content are blue and often underlined. Our CSS uses orange and an underline for links. It would be great if we could use the same style for the links within the diagram. I'm not sure bold is the right choice to indicate a clickable link. 

- _Create Expectations_ should be updated to _Add Expectations_ to match the convention in the diagram. Alternatively, you can update the convention in the workflow diagram to match the topic heading.

- I would remove _in this quickstart_ from _The following diagram illustrates the end-to-end GX data validation workflow used in this quickstart_ as not all of the workflow steps appear in the quickstart as distinct headings. For example, _Run a Checkpoint_ is part of _Validate data_.  You could break the topics into sections to fix this issue, or consolidate your workflow steps.  Minor nit. I'll leave it with you to determine if you want to resolve it.

Unblocking to allow you to merge when ready.

**Review by kwcanuck — APPROVED** [review]

**Pull request merged**